src/transformers/sparse.py

```diff
@@ -0,0 +1,241 @@
+import collections
+import inspect
+import math
+import os
+from typing import Optional
+
+import numpy
+import torch
+
+import onnxruntime
+from sparseml.pytorch.optim import LayerPruningModifier, QuantizationModifier
+from sparseml.pytorch.optim.manager import ScheduledModifierManager
+from sparseml.pytorch.optim.optimizer import ScheduledOptimizer
+from sparseml.pytorch.utils import logger
+from transformers import Trainer
+from transformers.file_utils import RECIPE_NAME, WEIGHTS_NAME
+
+
+class SparseMLTrainer(Trainer):
```

Comment:
What was the reasoning for adding this class? Was this functionality not here before?

Reply:
This is a file I noted for myself to remove before pushing and did not follow through